Dockerfile

```diff
@@ -11,10 +11,18 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-FROM golang:1.9
+FROM golang:1.9-alpine3.6 AS builder
 
-WORKDIR /go/src/github.com/awslabs/amazon-ecr-credential-helper
+COPY ecr-login /go/src/github.com/awslabs/amazon-ecr-credential-helper/ecr-login
 
-COPY . .
+RUN env CGO_ENABLED=0 go install -installsuffix "static" \
+    github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login
 
-CMD make
+FROM alpine:3.6
```

Comment:
Is there any particular reason you need a full Alpine container instead of a scratch container?  The executable is statically-linked, so the only thing that should be necessary at that point is a CA certificate bundle rather than the rest of the Linux userland.

Reply:
Internally we used this container to run scripts within our CI so we found it useful to have basic UNIX utilities. Alpine overhead is around 4 MB which seemed reasonable to me. Also I am not sure how to just add the CA certificate bundle. If you have some pointers, I shall be happy to change this.

Reply:
I think my preference would be for a scratch container so that we don't need to worry about updating the base layer.  You can see an example that builds a scratch container with a CA certificate bundle [here](https://github.com/jahkeup/updater53/blob/8a7f8bcd978d06241b96e0876fcc12c9c1d93d64/Dockerfile).

Reply:
OK I shall update the Dockerfile to use scratch base.